common/tests/gballoc_ll_mimalloc_int/gballoc_ll_mimalloc_int.c

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#ifdef __cplusplus
+#include <cstdlib>
+#include <cstddef>
+#else
+#include <stdlib.h>
+#include <stddef.h>
+#endif
+
+#include "azure_macro_utils/macro_utils.h"
+#include "testrunnerswitcher.h"
+
+static TEST_MUTEX_HANDLE g_testByTest;
+
+#include "azure_c_pal/gballoc_ll.h"
+
+BEGIN_TEST_SUITE(gballoc_ll_mimalloc_int)
+
+TEST_SUITE_INITIALIZE(TestClassInitialize)
+{
+    g_testByTest = TEST_MUTEX_CREATE();
+    ASSERT_IS_NOT_NULL(g_testByTest);
+}
+
+TEST_SUITE_CLEANUP(TestClassCleanup)
+{
+    TEST_MUTEX_DESTROY(g_testByTest);
+}
+
+TEST_FUNCTION_INITIALIZE(TestMethodInitialize)
+{
+    if (TEST_MUTEX_ACQUIRE(g_testByTest))
+    {
+        ASSERT_FAIL("our mutex is ABANDONED. Failure in test framework");
+    }
+}
+
+TEST_FUNCTION_CLEANUP(TestMethodCleanup)
+{
+    TEST_MUTEX_RELEASE(g_testByTest);
+}
+
+
+TEST_FUNCTION(gballoc_ll_init_works)
+{
+    ///act
+    gballoc_ll_init(NULL);
+
+    ///assert - doesn't crash
+}
+
+TEST_FUNCTION(gballoc_ll_deinit_works)
+{
+    ///act
+    gballoc_ll_deinit();
+
+    ///assert - doesn't crash
+}
+
+TEST_FUNCTION(gballoc_ll_malloc_works)

```

Comment:
just for sanity, test alloc a 1MB buffer too

Reply:
ok

---
In reply to: [457701994](https://github.com/Azure/azure-c-pal/pull/11#discussion_r457701994) [](ancestors = 457701994)

Reply:
added here and in all the int tests.

---
In reply to: [457761934](https://github.com/Azure/azure-c-pal/pull/11#discussion_r457761934) [](ancestors = 457761934,457701994)